app/src/main/java/com/example/learn09082021v001/MainActivity.kt

```diff
@@ -1,39 +1,66 @@
 package com.example.learn09082021v001
 
+import android.app.Activity
 import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import kotlinx.android.synthetic.main.activity_data_input.*
 import kotlinx.android.synthetic.main.activity_main.*
+
+
 class MainActivity : AppCompatActivity() {
+    override fun onSaveInstanceState(savedInstanceState: Bundle) {
+        super.onSaveInstanceState(savedInstanceState)
+        savedInstanceState.putString("nameandroidView", tvViewNameAndroid.text.toString())
+        savedInstanceState.putString("scoreandroidView", tvViewScoreAndroid.text.toString())
+        savedInstanceState.putString("nameiosView", tvViewNameIOS.text.toString())
+        savedInstanceState.putString("scoreiosView", tvViewScoreIOS.text.toString())
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        tvViewNameAndroid.text = savedInstanceState.getString("nameandroidView")
+        tvViewScoreAndroid.text = savedInstanceState.getString("scoreandroidView")
+        tvViewNameIOS.text = savedInstanceState.getString("nameiosView")
+        tvViewScoreIOS.text = savedInstanceState.getString("scoreiosView")
+
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (data != null && resultCode == RESULT_OK) {
+            tvViewNameAndroid.setText(data.getStringExtra("nameandroidSend"))
+            tvViewNameIOS.setText(data.getStringExtra("nameiosSend"))
+            tvViewScoreAndroid.setText(data.getStringExtra("scoreandroidSend"))
+            tvViewScoreIOS.setText(data.getStringExtra("scoreiosSend"))
+            datacheck()
```

Comment:
call `datacheck` here is redundant. If necessary to validate data, it should be called before set data to views

Reply:
datacheck funtion : it is check data empty or data not empty : will settext btn >> START OR RESET